source/administration/incident-response-application.rst

```diff
@@ -0,0 +1,667 @@
+
+For example, you can specify "#s1critical" as the text trigger and any message that contains this keyword is treated as an incident and starts the workflow.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "team_name", "The team(s) to watch for the trigger. If none specified, all teams will be watched.", "string", "No"
+    "channel_name", "The channel(s) to look for the trigger. If none specified, all channels will be watched.", "string", "No"
+    "match", "The text string to match on", "string", "If ``match_regex`` not set, yes"
+    "match_regex", "The regex to match on", "string", "If ``match`` not set, yes"
+
+Below is an example JSON for a text trigger where any message posted to Nagios channel in InfoSec team containing the word ``#s1critical`` triggers the workflow:
+
+.. code-block:: json
+
+  "triggers": [
+      {
+          "name": "IssueReported",
+          "type": "text",
+          "team_name": "infosec",
+          "channel_name": "nagios",
+          "match": "#s1critical",
+          "match_regex": ""
+      }
+  ]
+
+Step
+^^^^^^^^^^^^^^^
+
+A step is the representation of a state the app workflow can be in. For now it contains a set of actions to be performed when it is started and when it is finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the step", "string", "Yes. This field must be unique."
+    "start_actions", "Actions to perform when the step is reached.", "[]Action", "If ``finished_actions`` not set, yes"
+    "finish_actions", "Actions to perform when the step is finished.", "[]Action", "If ``start_actions`` not set, yes"
+
+Note that the step name must be a sequence of one or more letters, digits and underscores, and start with a letter.
+
+Action
+^^^^^^^^^^^^^^^
+
+Actions are performed when steps are started and finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the action", "string", "Yes. This field must be unique."
+    "type", "The type of action to perform", "string", "Yes"
+    "*additional fields", "Additional fields based on the type of action. See below for more details.", "string", "Yes"
+
+Note that the action name must be a sequence of one or more letters, digits and underscores, and start with a letter.
+
+Create Channel (type: ``create_channel``)
+*******************************************
+
+Creates a channel with the given parameters.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel_name", "The name of the channel. This is the channel handle used in the URL.", "string", "Yes"
+    "channel_displayname", "The display name of the channel.", "string", "Yes"
+    "team_name", "The team to create the channel in. If none specified, the channel is created in the same team as where the app workflow was triggered.", "string", "No"
```

Comment:
@crspeller Is `If none specified, the channel is created in the same team as where the app workflow was triggered.` correct? Or if none specified, is it created in the team where the action was taken?

Reply:
It's where the workflow was triggered.